core/src/main/scala/kafka/log/Log.scala

```diff
@@ -576,7 +576,11 @@ class Log(@volatile var dir: File,
     lock synchronized {
       maybeHandleIOException(s"Error while renaming dir for $topicPartition in log dir ${dir.getParent}") {
         val renamedDir = new File(dir.getParent, name)
-        closeHandlers()
+
+        if(OperatingSystem.IS_WINDOWS){
+          close()
```

Comment:
1. It may take a while before the .deleted directory to be deleted; 
2. I changed to from "closeHandler()" to "close()" because we don't need to set the "isMemoryMappedBufferClosed" to true since the directory will be reopened. And because the issue you mentioned, it will prevent the .deleted directory from being deleted. And Close() is safer - it will try to append new log before it is closed.